File: Latihan2/C++/Hardware.cpp
#include <string>
#include <iostream>
using namespace std;
//import class Product
#include "Product.cpp"

//kelas yang digunakan untuk mengimplementasikan Hardware, turunan dari kelas Product
class Hardware : public Product {
private:
    //private attribute dari kelas Hardware
    string brand;
    string model;

public:
    //constructor
    Hardware() {}

    //mengeset nilai atribut brand
    void setBrand(string brand) {
        this->brand = brand;
    }

    //return nilai atribut brand
    string getBrand() {
        return this->brand;
    }

    //mengeset nilai atribut model
    void setModel(string model) {
        this->model = model;
    }

    //return nilai atribut model
    string getModel() {
        return this->model;
    }

    //menampilkan atribut dari Hardware
    void printHardware() {
        cout << "Brand : " << this->getBrand() << endl;
        cout << "Model : " << this->getModel() << endl;
    }

    //destructor
    ~Hardware() {}
};

File: Latihan2/Python/Hardware.py
#mengimport kelas Product
from Product import Product

#kelas yang digunakan untuk mengimplementasikan Hardware, turunan dari kelas Product
class Hardware(Product):
    def __init__(self):
        #constructor
        self.brand = ""
        self.model = ""

    def setBrand(self, brand):
        #mengeset nilai atribut brand
        self.brand = brand

    def getBrand(self):
        #return nilai atribut brand
        return self.brand

    def setModel(self, model):
        #mengeset nilai atribut model
        self.model = model

    def getModel(self):
        #return nilai atribut model
        return self.model

    def printHardware(self):
        #menampilkan atribut dari Hardware
        print("Brand: " + self.getBrand())
        print("Model: " + self.getModel())
